src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsF2Uap.java

```diff
@@ -0,0 +1,93 @@
+import com.wikia.webdriver.common.core.drivers.Browser;
+import com.wikia.webdriver.common.core.helpers.Emulator;
+import com.wikia.webdriver.common.dataprovider.ads.F2AdsDataProvider;
+import com.wikia.webdriver.common.templates.fandom.AdsF2TestTemplate;
+import com.wikia.webdriver.pageobjectsfactory.pageobject.adsbase.AdsFandomObject;
+
+import org.testng.annotations.Test;
+
+public class TestAdsF2Uap extends AdsF2TestTemplate {
+
+  @Test(
+      dataProviderClass = F2AdsDataProvider.class,
+      dataProvider = "fandomArticleUapPage",
+      groups = {"AdsF2UapDesktop"}
+  )
+  public void adsFandomArticleUapDesktop(String pageType, String pageName, long atfId, long btfId) {
+    AdsFandomObject fandomPage = loadPage(pageName, pageType);
+    fandomPage.triggerOnScrollSlots();
+    verifyUapAtf(atfId, AdsFandomContent.TOP_BOXAD, fandomPage);
+    verifyUapBtf(btfId, AdsFandomContent.INCONTENT_BOXAD, AdsFandomContent.BOTTOM_LEADERBOARD, fandomPage);
+  }
+
+  @InBrowser(
+      browser = Browser.CHROME,
+      emulator = Emulator.GOOGLE_NEXUS_5
+  )
+  @Test(
+      dataProviderClass = F2AdsDataProvider.class,
+      dataProvider = "fandomArticleUapPage",
+      groups = {"AdsF2UapMobile"}
+  )
+  @RelatedIssue(issueID = "ADEN-4339")
+  public void adsFandomArticleUapMobile(String pageType, String pageName, long atfId, long btfId) {
+    AdsFandomObject fandomPage = loadPage(pageName, pageType);
+    fandomPage.triggerOnScrollSlots();
+    
+    verifyUapAtf(atfId, AdsFandomContent.TOP_BOXAD, fandomPage);
+    fandomPage.triggerOnScrollSlots();
+    verifyUapBtf(btfId, AdsFandomContent.INCONTENT_BOXAD, AdsFandomContent.BOTTOM_LEADERBOARD, fandomPage);
+  }
+
+  @Test(
+      dataProviderClass = F2AdsDataProvider.class,
+      dataProvider = "fandomTopicPage",
+      groups = {"AdsF2UapDesktop"}
+  )
+  public void adsF2UapOnTopicDesktop(String pageType, String pageName, long atfId, long btfId) {
+    AdsFandomObject fandomPage = loadPage(pageName, pageType);
+    fandomPage.triggerOnScrollSlots();
+    verifyUapAtf(atfId, AdsFandomContent.TOP_BOXAD, fandomPage);
+    verifyUapBtf(btfId, AdsFandomContent.FEED_BOXAD, AdsFandomContent.BOTTOM_LEADERBOARD, fandomPage);
+  }
+
+  @InBrowser(
+      browser = Browser.CHROME,
+      emulator = Emulator.GOOGLE_NEXUS_5
+  )
+  @Test(
+      dataProviderClass = F2AdsDataProvider.class,
+      dataProvider = "fandomTopicPage",
+      groups = {"AdsF2UapMobile"}
+  )
+  @RelatedIssue(issueID = "ADEN-4339")
+  public void adsF2UapOnTopicMobile(String pageType, String pageName, long atfId, long btfId) {
+    AdsFandomObject fandomPage = loadPage(pageName, pageType);
+    fandomPage.triggerOnScrollSlots();
+    verifyUapAtf(atfId, AdsFandomContent.TOP_BOXAD, fandomPage);
+    verifyUapBtf(btfId, AdsFandomContent.FEED_BOXAD, AdsFandomContent.BOTTOM_LEADERBOARD, fandomPage);
+  }
+
+  private void verifyUapAtf(long atfId, String slotName, AdsFandomObject fandomPage) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused private "verifyUapAtf" method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)